ng2-components/ng2-alfresco-upload/src/components/upload-button.component.ts

```diff
@@ -158,39 +163,35 @@ export class UploadButtonComponent implements OnInit, OnChanges {
-            if (hasPermission) {
-                let hashMapDir = this.convertIntoHashMap(files);
-
-                hashMapDir.forEach((filesDir, directoryPath) => {
-                    let directoryName = this.getDirectoryName(directoryPath);
-                    let absolutePath = this.currentFolderPath + this.getDirectoryPath(directoryPath);
-
-                    this.uploadService.createFolder(absolutePath, directoryName, this.rootFolderId)
-                        .subscribe(
-                            res => {
-                                let relativeDir = this.currentFolderPath + '/' + directoryPath;
-                                this.uploadFiles(relativeDir, filesDir);
-                            },
-                            error => {
-                                let errorMessagePlaceholder = this.getErrorMessage(error.response);
-                                if (errorMessagePlaceholder) {
-                                    this.onError.emit({value: errorMessagePlaceholder});
-                                    let errorMessage = this.formatString(errorMessagePlaceholder, [directoryName]);
-                                    if (errorMessage) {
-                                        this._showErrorNotificationBar(errorMessage);
-                                    }
+        if (this.hasPermission) {
+            let hashMapDir = this.convertIntoHashMap(files);
+
+            hashMapDir.forEach((filesDir, directoryPath) => {
+                let directoryName = this.getDirectoryName(directoryPath);
+                let absolutePath = this.currentFolderPath + this.getDirectoryPath(directoryPath);
+
+                this.uploadService.createFolder(absolutePath, directoryName, this.rootFolderId)
+                    .subscribe(
+                        res => {
+                            let relativeDir = this.currentFolderPath + '/' + directoryPath;
+                            this.uploadFiles(relativeDir, filesDir);
+                        },
+                        error => {
+                            let errorMessagePlaceholder = this.getErrorMessage(error.response);
+                            if (errorMessagePlaceholder) {
+                                this.onError.emit({value: errorMessagePlaceholder});
+                                let errorMessage = this.formatString(errorMessagePlaceholder, [directoryName]);
+                                if (errorMessage) {
+                                    this._showErrorNotificationBar(errorMessage);
                                 }
-                                this.logService.error(error);
                             }
-                        );
-                });
-            } else {
-                this.permissionEvent.emit(new PermissionModel({type: 'content', action: 'upload', permission: 'create'}));
-            }
-        });
-
-        this.checkPermission();
+                            this.logService.error(error);
```

Comment:
the log of the error is already made by the service you cane emit an error event